airflow/serialization/serialized_objects.py

```diff
@@ -283,7 +305,17 @@ def decode_asset_condition(var: dict[str, Any]) -> BaseAsset:
     """
     dat = var["__type"]
     if dat == DAT.ASSET:
-        return Asset(name=var["name"], uri=var["uri"], group=var["group"], extra=var["extra"])
+        serialized_watchers = var["watchers"] if "watchers" in var else []
+        return Asset(
+            name=var["name"],
+            uri=var["uri"],
+            group=var["group"],
+            extra=var["extra"],
```

Comment:
Consistency please -- There is another deserialize (why are there two? Separate question I guess) but 
```suggestion
            **var,
```

Reply:
I dont have the answer for this one. What I know is one if at the root level of `deserialize` (then deserializing an asset) and the other one is in `decode_asset_condition` so deserializing an asset condition. I can create a private function to factorize code if you like it better

Reply:
I don't mind two versions per se, but it'd be nice if they had the same implementation 😁

Reply:
I applied this one but reverted because `var` contains some keys we do not want to pass to `Asset`. See error [here](https://github.com/apache/airflow/actions/runs/12914159501/job/36013477933?pr=45562)

Reply:
I factorized the code in a new function `decode_asset`